evennia/contrib/tree_select.py

```diff
@@ -0,0 +1,522 @@
+the player to select between those options, or to go back to the previous menu. In
+addition, categories are marked by default with a '[+]' at the end of their key. Both
+this marker and the option to go back can be disabled.
+
+Categories can be nested in other categories as well - just go another '-' deeper. You
+can do this as many times as you like. There's no hard limit to the number of
+categories you can go down.
+
+For example, let's add some more options to our menu, turning 'Bar' into a category.
+
+    TEST_MENU = '''Foo
+    Bar
+    -You've got to know
+    --When to hold em
+    --When to fold em
+    --When to walk away
+    Baz
+    Qux'''
+    
+Now when we call the menu, we can see that 'Bar' has become a category instead of a
+selectable option.
+
+    _______________________________
+
+    Make your selection:
+    _______________________________
+
+    Foo      
+    Bar [+]  
+    Baz      
+    Qux      
+    
+Note the [+] next to 'Bar'. If we select 'Bar', it'll show us the option listed under it.
+
+    ________________________________________________________________
+
+    Bar
+    ________________________________________________________________
+    
+    You've got to know [+]                    
+    << Go Back: Return to the previous menu.  
+    
+Just the one option, which is a category itself, and the option to go back, which will
+take us back to the previous menu. Let's select 'You've got to know'.
+
+    ________________________________________________________________
+    
+    You've got to know
+    ________________________________________________________________
+    
+    When to hold em                           
+    When to fold em                           
+    When to walk away                         
+    << Go Back: Return to the previous menu.
+    
+Now we see the three options listed under it, too. We can select one of them or use 'Go
+Back' to return to the 'Bar' menu we were just at before. It's very simple to make a
+branching tree of selections!
+
+One last thing - you can set the descriptions for the various options simply by adding a
+':' character followed by the description to the option's line. For example, let's add a
+description to 'Baz' in our menu:
+
+    TEST_MENU = '''Foo
+    Bar
+    -You've got to know
+    --When to hold em
+    --When to fold em
+    --When to walk away
+    Baz: Look at this one: the best option.
+    Qux'''
+    
+Now we see that the Baz option has a description attached that's separate from its key:
+
+    _______________________________________________________________
+
+    Make your selection:
+    _______________________________________________________________
+    
+    Foo                                      
+    Bar [+]                                  
+    Baz: Look at this one: the best option.  
+    Qux                                      
+    
+And that's all there is to it! For simple branching-tree selections, using this system is
```

Comment:
There is one thing missing from this (otherwise excellent) example/docstring - and that is an example of the callback and how it is called with the result of the menu. It's mentioned at the top, but it might be useful to put a few lines about that here when everything else is described so explicitly.

Reply:
Done!